hibernate-core/src/main/java/org/hibernate/dialect/FirebirdDialect.java

```diff
@@ -594,4 +601,20 @@ public JDBCException convert(SQLException sqlException, String message, String s
 		};
 	}
 
+	@Override
+	public SqmMultiTableMutationStrategy getFallbackSqmMutationStrategy(EntityMappingType entityDescriptor, RuntimeModelCreationContext runtimeModelCreationContext) {
+		return getVersion() < 210
+				? super.getFallbackSqmMutationStrategy( entityDescriptor, runtimeModelCreationContext )
+				: new GlobalTemporaryTableStrategy(
+						new IdTable( entityDescriptor, name -> "HT_" + name ),
+						() -> new TempIdTableExporter( false, this::getTypeName ) {
+							@Override
+							protected String getCreateOptions() {
+								return "on commit delete rows";
+							}
+						},
+						AfterUseAction.CLEAN,
```

Comment:
Is this right?

Reply:
Not sure how firebird implements their global-temp-table support.  Cleaning it should be fine, its more a question of whether its needed.  I'd leave it as is